pilot/pkg/features/pilot.go

```diff
@@ -126,6 +126,12 @@ var (
 		"If not empty, services with this label will use cookie based persistent sessions",
 	).Get()
 
+	PersistentCookiePathLabel = env.Register(
+		"PILOT_PERSISTENT_COOKIE_PATH_LABEL",
+		"istio.io/persistent-cookie-path",
+		"Specifies optional cookie path for cookie based persistent sessions. '/' is default. Use '.' instead of '/' in the path.",
```

Comment:
This is the svc label? If so, should not be in istio/api instead of a feature env.

Reply:
This is an extension of other labels in that file (`PILOT_PERSISTENT_COOKIE_PATH_LABEL` and `PILOT_DRAINING_LABEL` etc). You can see few lines above and below.

For istio/api there is an existing PR https://github.com/istio/api/pull/2469 which underwent a lot of discussion and we needed something quicker.

Reply:
I did see that

Reply:
Sure, is there an example of an existing annotation I can look at?

Reply:
I mean annotations do not have validation,  if we just move from label to annotation we may be fine

Reply:
Are you saying if I use an annotation on the service it will show up in the exact same field [`svc.Attributes.Labels`](https://github.com/istio/istio/blob/master/pilot/pkg/model/service.go#L574) ? In that case I don't need any changes

Reply:
No. We need to change and pass annotations to svc.Attributes or atleast capture these three specific annotations and store in svc object as persistent session info.

What John is suggesting is if we switch from label -> annotation and make the above change, we do not need additional label.

Reply:
Thanks @ramaraochavali . Since I am new to the code it will be good if you give me some pointers as to how I can do it.

Reply:
I agree with @howardjohn @ramaraochavali and think that annotations in general better suit for the purpose here. In addition we also need to consider how cookie TTL configuration would be specified.